### PR TITLE
Restore custom db passwords on package remove and re-install and on u…

### PR DESCRIPTION
**Pull request by brianhatchl**
…pdate

### CODE REVIEW

File: src/SPECS/hootenanny.spec.in [hunk 1]
```diff
@@ -175,6 +175,12 @@ if [ "$1" = "2" ]; then
     if [ -d $SERVICES_HOME ]; then
         sudo rm -rf $SERVICES_HOME
     fi
+
+    # If DatabaseConfig.sh still has passwords, write these to DatabaseConfigLocal.sh
+    if grep -qw PASSWORD /var/lib/hootenanny/conf/DatabaseConfig.sh; then
+        sudo grep DB_PASSWORD /var/lib/hootenanny/conf/DatabaseConfig.sh.rpmsave > /var/lib/hootenanny/conf/DatabaseConfigLocal.sh
+    else
+
 fi
```
Comment (brianhatchl) [inline src/SPECS/hootenanny.spec.in]: This code is in the pre yum update section

File: src/SPECS/hootenanny.spec.in [hunk 2]
```diff
@@ -206,6 +212,15 @@ if [ "$1" = "1" ]; then
...
 
+    # restore saved db config file settings if present
+    if [ -f /var/lib/hootenanny/conf/DatabaseConfig.sh.rpmsave ]; then
+        if [ -f /var/lib/hootenanny/conf/DatabaseConfigDefault.sh ]; then
+            sudo grep DB_PASSWORD /var/lib/hootenanny/conf/DatabaseConfig.sh.rpmsave > /var/lib/hootenanny/conf/DatabaseConfigLocal.sh
+        else
+            sudo mv /var/lib/hootenanny/conf/DatabaseConfig.sh.rpmsave /var/lib/hootenanny/conf/DatabaseConfig.sh
+        fi
+    fi
+
```
Comment (brianhatchl) [inline src/SPECS/hootenanny.spec.in]: This code is in the post yum install section, but handles the case where a hoot package was removed and then installed again.

**Comment by mattjdnv** [comment]
Should we wait for the RPM test or just merge?

**Comment by brianhatchl** [comment]
I think we can just merge.  These fixes will only get exercised by `yum update` or `yum remove` then `yum install`.  This jenkins job only tests a clean install.

**Comment by mattjdnv** [comment]
Done.

**Pull request merged**